example/ck_tile/01_fmha/fmha_fwd.hpp

```diff
@@ -145,6 +145,8 @@ struct fmha_fwd_args
     float p_drop;
     bool s_randval;
     std::tuple<uint64_t, uint64_t> drop_seed_offset;
+    const void* drop_seed_ptr;
```

Comment:
I think `drop_seed_offset` are kernel arguments located in host side; and `drop_seed_ptr` & `drop_offset_ptr` are device side arguments. We can make a `union` to choose between these 2 options.

Reply:
We may create something like `drop_seed_offset_t` (shown below) to accommodate either `uint64` & pointer kernel arguments. This will save up to 8-byte sgpr compare to current approach. 

```c++
struct drop_seed_offset_t
{
    union {
        struct {
            const void* seed_ptr;
            const void* offset_ptr;
        } device;

        struct {
            std::uint64_t seed;
            std::uint64_t offset;
        } host;
    } payload;

    bool is_host;
};

static_assert(sizeof(drop_seed_offset_t) == 2 * 8 + 8);

auto make_host_drop_seed_offset(std::uint64_t seed, std::uint64_t offset) -> drop_seed_offset_t {
    drop_seed_offset_t seed_offset;

    seed_offset.payload.host.seed = seed;
    seed_offset.payload.host.offset = offset;
    seed_offset.is_host = true;

    return seed_offset;
}
```